lib/Controller/AccountsController.php

```diff
@@ -131,13 +130,48 @@ public function show($accountId) {
 		}
 	}
 
+	/**
+	 * @param array $config
+	 * @param integer $id
+	 */
+	private function createAccount($config, $id = null) {
+		$account = null;
+		$autoDetect = $config['autoDetect'] === 'true';
+
+		try {
+			if ($autoDetect) {
+				$account = $this->setup->createNewAutoconfiguredAccount($config['accountName'], $config['emailAddress'], $config['password']);
+				if (is_null($account)) {
+					throw new Exception();
+				}
+			} else {
+				$account = $this->setup->createNewAccount($config['accountName'], $config['emailAddress'], $config['imapHost'], $config['imapPort'], $config['imapSslMode'], $config['imapUser'], $config['imapPassword'], $config['smtpHost'], $config['smtpPort'], $config['smtpSslMode'], $config['smtpUser'], $config['smtpPassword'], $this->currentUserId, $id);
+			}
+		} catch (Exception $ex) {
+			if ($autoDetect) {
+				return new JSONResponse([
+					'message' => $this->l10n->t('Auto detect failed. Please use manual mode.')
+					], Http::STATUS_BAD_REQUEST);
+			} else {
+				$this->logger->error('Creating account failed: ' . $ex->getMessage());
+				return new JSONResponse([
+					'message' => $this->l10n->t('Creating account failed: ') . $ex->getMessage()
+					], Http::STATUS_BAD_REQUEST);
+			}
+		}
+
+		return new JSONResponse(['data' => ['id' => $account->getId()]], Http::STATUS_CREATED);
+	}
+
 	/**
 	 * @NoAdminRequired
+	 *
+	 * @param int $id
+	 * @param array $config
+	 * @return JSONResponse
 	 */
-	public function update() {
-		$response = new Response();
-		$response->setStatus(Http::STATUS_NOT_IMPLEMENTED);
-		return $response;
+	public function update($id, $config) {
+		return $this->createAccount($config, $id);
```

Comment:
I'd say let's duplicate the method above instead of abusing it for the update. This kind of core re-use makes the code pretty hard to understand and thus might introduce new bugs.

Reply:
FYI the autodetected and the manual mode should be two separate routes and separate methods. This cleanup is out of scope for this PR however.